src/Form.jsx

```diff
@@ -114,15 +115,23 @@ function Form({
     const signatureRef = useRef({}).current;
     const maskedRef = useRef({}).current;
 
+    // Calculate how many repeated rows there are given the current field values
+    const repeatedRowCount = useMemo(
+        () =>
+            rawActiveStep
+                ? calculateRepeatedRowCount({
+                      step: rawActiveStep,
+                      values: fieldValues
+                  })
+                : null,
+        [rawActiveStep, repeatChanged]
+    );
+
     // Create the fully-hydrated activeStep by injecting repeated rows
     // and pre-calculating styles.
     // Note: Other hydration transformations can also be included here
     const activeStep = useMemo(() => {
```

Comment:
should you include repeatedRowCount as a dependency in this useMemo? Are there any race conditions introduced between these two usememos?